src/ports/postgres/modules/utilities/minibatch_preprocessing.py_in

```diff
@@ -0,0 +1,559 @@
+                contains NULL
+        2. Please keep the null checking where clause of this query in sync with
+        the query in _get_skipped_rows_processed_count. We are doing this null
+        check in two places to prevent another pass of the entire dataset.
+        """
+
+        # This ID is the unique row id that get assigned to each row after preprocessing
+        unique_row_id = "__id__"
+        sql = """
+            CREATE TABLE {output_table} AS
+            SELECT {row_id},
+                   {schema_madlib}.matrix_agg({dep_colname}) as {dep_colname},
+                   {schema_madlib}.matrix_agg({ind_colname}) as {ind_colname}
+            FROM (
+                SELECT (row_number() OVER (ORDER BY random()) - 1) / {buffer_size}
+                            as {row_id}, * FROM
+                (
+                    {standardize_query}
+                 ) sub_query_1
+                 WHERE NOT {schema_madlib}.array_contains_null({dep_colname})
+                 AND NOT {schema_madlib}.array_contains_null({ind_colname})
+            ) sub_query_2
+            GROUP BY {row_id}
+            {distributed_by_clause}
+            """.format(
+            schema_madlib=self.schema_madlib,
+            source_table=self.source_table,
+            output_table=self.output_table,
+            dependent_varname=self.dependent_varname,
+            independent_varname=self.independent_varname,
+            buffer_size = calculated_buffer_size,
+            dep_colname=MINIBATCH_OUTPUT_DEPENDENT_COLNAME,
+            ind_colname=MINIBATCH_OUTPUT_INDEPENDENT_COLNAME,
+            row_id = unique_row_id,
+            distributed_by_clause = '' if is_platform_pg() else 'DISTRIBUTED RANDOMLY',
+            **locals())
+        plpy.execute(sql)
+
+
+        standardizer.create_output_standardization_table()
+        MiniBatchSummarizer.create_output_summary_table(
+            self.source_table,
+            self.output_table,
+            self.dependent_varname,
+            self.independent_varname,
+            calculated_buffer_size,
+            dep_var_classes_str,
+            num_rows_processed,
+            num_missing_rows_skipped,
+            self.output_summary_table)
+
+    def _validate_minibatch_preprocessor_params(self):
+        # Test if the independent variable can be typecasted to a double precision
+        # array and let postgres validate the expression
+
+        # Note that this will not fail for 2d arrays but the standardizer will
+        # fail because utils_normalize_data will throw an error
+        typecasted_ind_varname = "{0}::double precision[]".format(
+                                                    self.independent_varname)
+        validate_module_input_params(self.source_table, self.output_table,
+                                     typecasted_ind_varname,
+                                     self.dependent_varname, self.module_name)
+
+        self._validate_other_output_tables()
```

Comment:
This is linked to the last comment on this PR. We could use an optional param in `validate_module_input_params` to validate other output tables too (a list of suffixes such as `['_summary', '_standardization']`).